test/unittest/tests/wrappers/item_wrapper.cpp

```diff
@@ -9,7 +9,7 @@
 #include "wiredtiger.h"
 #include "item_wrapper.h"
 
-item_wrapper::item_wrapper(const char *string) : _string(string)
+item_wrapper::item_wrapper(std::string const &string) : _string(string)
 {
     _item.data = _string.c_str();
```

Comment:
My reading is that the `WT_ITEM` this class wraps is at least conceptually mutable.
However modifying the memory returned by `c_str()` is undefined behavior *[string.accessors]* in the C++17 standard.

Reply:
Hi @marcbutler , you are correct, and I agree that is not currently made clear.
The `item_wrapper` as currently implemented is intended to wrap only constant, read-only strings. I've added a comment to `item_wrapper.h` explaining that, and we should consider enhancing `item_wrapper` in future tickets.

Reply:
agreed